domains/client/domain-operator/src/operator.rs

```diff
@@ -178,19 +189,136 @@ where
             domain_block_processor.clone(),
         );
 
+        let snap_sync_orchestrator = params
+            .consensus_chain_sync_params
+            .as_ref()
+            .map(|p| p.snap_sync_orchestrator.clone());
+
+        let sync_params = params
+            .consensus_chain_sync_params
+            .map(|consensus_sync_params| SyncParams {
+                domain_client: params.client.clone(),
+                domain_network_service_handle: params.domain_network_service_handle,
+                sync_service: params.sync_service,
+                consensus_client: params.consensus_client.clone(),
+                domain_block_downloader: params.block_downloader.clone(),
+                consensus_chain_sync_params: consensus_sync_params,
+                domain_fork_id: params.domain_fork_id,
+                receipt_provider: params.domain_execution_receipt_provider,
+            });
+
+        if let Some(sync_params) = sync_params {
+            let domain_sync_task = {
+                async move {
+                    let info = sync_params.domain_client.info();
+                    // Only attempt snap sync with genesis state
+                    // TODO: Support snap sync from any state once
+                    //  https://github.com/paritytech/polkadot-sdk/issues/5366 is resolved
+                    if info.best_hash == info.genesis_hash {
+                        info!("Starting domain snap sync...");
+
+                        let result = snap_sync(sync_params).await;
+
+                        match result {
+                            Ok(_) => {
+                                info!("Domain snap sync completed.");
+                            }
+                            Err(err) => {
+                                error!(%err, "Domain snap sync failed.");
+
+                                // essential task failed
+                                return;
+                            }
+                        };
+                    } else {
+                        debug!("Snap sync can only work with genesis state, skipping");
+
+                        // essential task failed
+                        return;
```

Comment:
In this case, should we print an info log and fall back to full sync or shut down to let the user specify full sync and restart? In the latter, we should also print a log to provide instructions on what to do.

Reply:
Rolling back to the full sync is not supported at the moment without the restart and DB wiping.

Reply:
Then we should print a warming or error log with details of what happened and instructions on what to do, otherwise the current impl user just sees the node crash due to an essential task failed

Reply:
There is an error message already. I added additional instructions.